spec/models/manageiq/providers/openshift/container_manager/refresher_spec.rb

```diff
@@ -127,6 +126,164 @@ def normal_refresh
+    expect(@project0.containers.count).to eq(0)
+    expect(@project0.container_definitions.count).to eq(0)
+  end
+
+  def assert_specific_pods_disconnected
+    @pod0 = ContainerGroup.find_by(:name => "my-pod", :old_container_project_id => @project0.id)
+    @pod1 = ContainerGroup.find_by(:name => "my-pod", :old_container_project_id => @project1.id)
+
+    [@pod0, @pod1].each do |pod|
+      assert_disconnected(pod)
+      expect(pod.container_project).to be_nil
+      expect(pod.containers.count).to eq(1)
+      expect(pod.container_definitions.count).to eq(1)
+    end
+  end
+
+  def assert_specific_container_definitions_disconnected
+    @container_def0 = ContainerDefinition.find_by(:name => "my-container", :container_group => @pod0)
+    @container_def1 = ContainerDefinition.find_by(:name => "my-container", :container_group => @pod1)
+
+    [@container_def0, @container_def1].each do |container_def|
+      assert_disconnected(container_def)
+      expect(container_def.container).not_to be_nil
+    end
+  end
+
+  def assert_specific_containers_disconnected
+    @container0 = Container.find_by(:name => "my-container", :container_definition => @container_def0)
+    @container1 = Container.find_by(:name => "my-container", :container_definition => @container_def1)
+
+    [@container0, @container1].each do |container|
+      assert_disconnected(container)
+      expect(container.container_project).to be_nil
+    end
+  end
+
+  def assert_disconnected(object)
+    expect(object).not_to be_nil
+    expect(object.deleted_on).not_to be_nil
+    expect(object.ext_management_system).to be_nil
+    expect(object.old_ems_id).to eq(@ems.id)
+  end
+
+  def assert_table_counts_before_deletions
+    expect(ContainerProject.count).to eq(8)
+    expect(ContainerGroup.count).to eq(10)
+    expect(Container.count).to eq(10)
+    expect(ContainerDefinition.count).to eq(10)
+    expect(ContainerImage.count).to eq(45)
+    expect(ContainerService.count).to eq(11)
+    expect(ContainerRoute.count).to eq(6)
+    expect(ContainerQuota.count).to eq(3)
+    expect(ContainerQuotaItem.count).to eq(15)
+    expect(ContainerLimit.count).to eq(3)
+    expect(ContainerLimitItem.count).to eq(12)
+    expect(ContainerTemplate.count).to eq(30)
+    expect(PersistentVolumeClaim.count).to eq(4)
+    expect(CustomAttribute.count).to eq(535)
+    expect(ContainerTemplateParameter.count).to eq(264)
+    expect(ContainerRoute.find_by(:name => "my-route", :container_project => @project2).labels.count).to eq(1)
+    expect(ContainerTemplate.find_by(:name => "my-template", :container_project => @project2)
+             .container_template_parameters.count).to eq(1)
+  end
+
+  def assert_table_counts_after_deletions
+    expect(ContainerProject.count).to eq(8)
+    expect(ContainerGroup.count).to eq(10)
+    expect(Container.count).to eq(10)
+    expect(ContainerDefinition.count).to eq(10)
+    expect(ContainerImage.count).to eq(44) # BUG?
```

Comment:
It is a little late now, but @zeari Do you know if the this the expected behavior?